infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesPersistentVolumeClaims.java

```diff
@@ -231,10 +218,56 @@ public void onClose(KubernetesClientException cause) {
       }
     } catch (KubernetesClientException e) {
       throw new KubernetesInfrastructureException(e);
-    } finally {
-      if (watch != null) {
-        watch.close();
-      }
     }
   }
+
+  private Watch pvcIsBoundWatcher(
+      CompletableFuture<PersistentVolumeClaim> future,
+      Resource<PersistentVolumeClaim, DoneablePersistentVolumeClaim> pvcResource,
+      Predicate<PersistentVolumeClaim> predicate,
+      String name) {
+    return pvcResource.watch(
+        new Watcher<PersistentVolumeClaim>() {
+          @Override
+          public void eventReceived(Action action, PersistentVolumeClaim pvc) {
+            if (predicate.test(pvc)) {
+              future.complete(pvc);
+            }
+          }
+
+          @Override
+          public void onClose(KubernetesClientException cause) {
+            future.completeExceptionally(
+                new InfrastructureException(
+                    "Waiting for persistent volume claim '" + name + "' was interrupted"));
+          }
+        });
+  }
+
+  private Watch pvcIsWaitingForConsumerWatcher(
+      CompletableFuture<PersistentVolumeClaim> future, PersistentVolumeClaim actualPvc)
+      throws InfrastructureException {
+    return clientFactory
+        .create(workspaceId)
+        .events()
+        .inNamespace(namespace)
+        .withField("reason", "WaitForFirstConsumer")
+        .withField("involvedObject.uid", actualPvc.getMetadata().getUid())
+        .watch(
+            new Watcher<Event>() {
+              private final Logger log = LoggerFactory.getLogger(getClass());
```

Comment:
Consider using adding static logger field to `KubernetesPersistentVolumeClaims` instead of creating new one for each event watcher

Reply:
good catch, fixed.